app/javascript/widget/views/Home.vue

```diff
@@ -50,6 +53,7 @@ export default {
       const ref = new URLSearchParams(window.location.search).get('referral');
       if (ref) {
         this.clearConversations();
+        await setCustomAttributes({"ref": ref});
```

Comment:
I believe we need to make the `startConversation` fn an async function so that we can use the await keyword

```suggestion
        methods: {
    ...mapActions('conversation', [
      'clearConversations',
    ]),
    async startConversation() { // Mark the method as async
      const ref = new URLSearchParams(window.location.search).get('referral');
      if (ref) {
        this.clearConversations();
        try {
          await setCustomAttributes({"ref": ref});
        } catch (error) {
          console.error('Error setting custom attributes:', error);
          // Handle error if necessary
        }
      } else if (this.preChatFormEnabled && !this.conversationSize) {
        return this.replaceRoute('prechat-form');
      }
      return this.replaceRoute('messages');
    },
  },

```